src/index.rst

```diff
@@ -616,6 +649,61 @@ Now ready to submit the workflow: ::
 
    $> bps submit test_sdf.yaml
 
+Submit a workflow to USDF (Developer)
+-------------------------------------
+
+For developer to submit a workflow to S3DF with local software in addition to an LSST stack, please at first check `Submit a workflow to USDF`_.
+Here we only list the differences.
+
+Copy the environment setup script from cvmfs to your local directory and update the lsst
+setup part to your private repo: ::
+
+   $> latest=$(ls -td /cvmfs/sw.lsst.eu/linux-x86_64/panda_env/v* | head -1)
+   $> cp $latest/setup_panda_s3df.sh /local/directory/
+   $> <update /local/directory/setup_panda_s3df.sh>
+   $> source /local/directory/setup_panda_s3df.sh
+
+``Note``: Make sure PanDA can read your private repo: ::
+
+   $> chmod -R g+rxs <your private development repo>
+
+For the submission yaml file ``test_usdf.yaml``, you need to change the ``runnercommand`` to point to your private development repo: ::
+
+   $> cat test_usdf.yaml
+   # An example bps submission yaml
+   # Need to setup USDF before submitting the yaml
+   # source setupUSDF.sh
+
+   LSST_VERSION: w_2022_32
+
+   includeConfigs:
+   - ${CTRL_BPS_PANDA_DIR}/config/bps_usdf.yaml
+
+   pipelineYaml: "${DRP_PIPE_DIR}/pipelines/HSC/DRP-RC2.yaml#isr"
+
+   payload:
+     payloadName: testUSDF
+     inCollection: "HSC/RC2/defaults"
+     dataQuery: "exposure = 34342 AND detector = 10"
+
+   # To replace 'loadLSST.bash' and 'setup lsst_distrib' with your local setup script..
+   runnerCommand: >
+      unset PYTHONPATH;
+      # source /cvmfs/sw.lsst.eu/linux-x86_64/lsst_distrib/{LSST_VERSION}/loadLSST.bash;
+      source /path/to/your/test/package/setup/script;
+      pwd; ls -al;
+      # setup lsst_distrib;
```

Comment:
The usual approach is that the user wants lsst_distrib set up and then wants their local setup below. This line can't be commented out.